File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        n=len(height)
        l=0
        r=n-1
        maximum=0
        ans=0
        while(l<r):
            ans=(r-l)*min(height[l],height[r])
            maximum=max(maximum,ans)
            if(height[l]<height[r]):
                l+=1
            else:
                r-=1
        return maximum

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int l=0,r=n-1;
        int maximum=0,ans=0;
        while(l<r){
            ans=(r-l)*min(height[l],height[r]);
            maximum=max(maximum,ans);
            if(height[l]<height[r]){
                l++;
            }else{
                r--;
            }
        }
        return maximum;
    }
};